test/support/ast_data.ex

```diff
@@ -154,7 +154,20 @@ defmodule Gradient.AstData do
     {expression, _} =
       :erl_parse.mapfold_anno(
         fn anno, acc ->
-          {{:erl_anno.line(anno) - acc, :erl_anno.column(anno)}, acc}
+          line =
+            case :erl_anno.line(anno) - acc do
+              line when is_integer(line) and line >= 0 ->
+                line
+            end
+
+          column =
+            case :erl_anno.column(anno) do
+              column when is_integer(column) and column > 0 ->
+                column
+            end
+
+          location = {line, column}
+          {location, acc}
```

Comment:
I think I've found a bug here: if instead of the above we do `{{line, column}, acc}`, we get this error:

```elixir
test/support/ast_data.ex:169: The tuple is expected to have type anno() but it has type {any(), any()}
167             end
168
169           {{line, column}, acc}
170         end,
171         :erl_anno.line(elem(expression, 1)),
```